x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/common/EmbeddingRequestChunker.java

```diff
@@ -177,9 +215,31 @@ public void onResponse(InferenceServiceResults inferenceServiceResults) {
                 int start = 0;
                 for (var pos : positions) {
                     results.get(pos.inputIndex())
-                        .setOnce(pos.chunkIndex(), textEmbeddingResults.embeddings().subList(start, start + pos.embeddingCount()));
+                        .setOnce(pos.chunkIndex(), embeddingResults.embeddings().subList(start, start + pos.embeddingCount()));
                     start += pos.embeddingCount();
                 }
+
+                if (firstResultType.get() == null) {
```

Comment:
Should we use `compareAndSet`? If two threads run this method, it's possible `firstResultType` gets set twice (making it the second result type?). Or if we don't want to run the `instanceof` checks every time, maybe something like:
```java
if(firstResultType.updateAndGet(currentValue -> {
    if(currentValue != null) {
        return currentValue;
    }
    if (inferenceServiceResults instanceof TextEmbeddingResults) {
        return EmbeddingType.FLOAT;
    } else if (inferenceServiceResults instanceof TextEmbeddingByteResults) {
        return EmbeddingType.BYTE;
    } else if (inferenceServiceResults instanceof SparseEmbeddingResults) {
        return EmbeddingType.SPARSE);
    }
    // what gets returned here? null?
}).matchedClass().equals(inferenceServiceResults.getClass()) == false)`
```
That is one `if` check lol

Reply:
Thanks that is a very good point that the update is not thread safe. 

I found it cleaner to add the `EmbeddingType` to the response interface so the code just asks the class now and avoids the big if statement. It also solves the problem of `EmbeddingType` being null as it can't be null now